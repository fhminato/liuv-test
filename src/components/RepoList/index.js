import React, {useState} from 'react';
import GitRepo from './../GitRepo';
import {FlatList} from 'react-native';

export default function RepoList(props) {
  return (
    <FlatList
      style={{width: '100%'}}
      data={props.data}
      renderItem={({item}) => (
        <GitRepo item={item} onClick={(item) => props.onClick(item)} />
      )}
      keyExtractor={(item, index) => `${index}`}></FlatList>
  );
}
