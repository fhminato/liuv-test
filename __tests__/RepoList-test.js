import React from 'react';
import RepoList from '../src/components/RepoList';

import renderer from 'react-test-renderer';

const repos = [
  {
    name: 'Test',
    language: 'React Native',
  },
];

test('renders correctly', () => {
  const tree = renderer
    .create(<RepoList data={repos} onClick={(item) => console.log(item)} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
