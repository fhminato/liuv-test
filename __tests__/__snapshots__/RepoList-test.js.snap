// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "language": "React Native",
        "name": "Test",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "width": "100%",
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderColor": "#777",
              "borderRadius": 5,
              "borderStyle": "solid",
              "borderWidth": 1,
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "display": "flex",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Antipasto Pro",
                  "textTransform": "uppercase",
                },
              ]
            }
          >
            Test
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Antipasto Pro",
                  "textTransform": "uppercase",
                },
              ]
            }
          >
            React Native
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#5000ca",
              "borderRadius": 100,
              "height": 25,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Antipasto Pro",
                  "fontSize": 11,
                  "textTransform": "uppercase",
                },
              ]
            }
          >
            Detalhes
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
